### PR TITLE
Calculate DesktopGameWindow.Position relative to the current screen

### LINKED ISSUE #1901: Restoring window position does not work on multi-monitor set up
Environment:
Windows 10 computer with 2 displays.

Steps to reproduce:
1. Launch visual-tests in windowed mode.
2. Move it to the secondary display.
3. Switch to fullscreen/borderless.
4. Switch back to windowed.

Expected results:
Window position is the same as before
before

Actual results:
Window position was changed. The window is on the primary display.
after

Additional info:
This code is the culprit, but it can't be just removed without testing.
culprit

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/63319241-restoring-window-position-does-not-work-on-multi-monitor-set-up?utm_campaign=plugin&utm_content=tracker%2F44024461&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F44024461&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #1908: DesktopGameWindow.Position does not support multi-monitor setups
Discovered this while fixing #1833.

Afaict, the `DesktopGameWindow.Position` property is supposed to indicate in a range from (0,0) to (1,1), where on the current screen the window is located.
While it does use the current display resolution for its calculations, the returned value is always absolute, not relative to the current screen. This results in values like shown in the screenshot below.
Another side effect is that a window will always "snap back" to the primary screen when windowed mode is entered.

position

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/63443881-desktopgamewindow-position-does-not-support-multi-monitor-setups?utm_campaign=plugin&utm_content=tracker%2F44024461&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F44024461&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by AtomCrafty**
Closes #1908
Closes #1901

### CODE REVIEW

File: SampleGame/SampleGame.cs [hunk 1]
```diff
@@ -8,26 +8,21 @@
 using osu.Framework.Graphics.Shapes;
 using osu.Framework.Allocation;
 
-namespace SampleGame
-{
-    internal class SampleGame : Game
-    {
+namespace SampleGame {
```
Comment (peppy) [inline SampleGame/SampleGame.cs]: please revert the changes in this file.

Comment (AtomCrafty) [reply SampleGame/SampleGame.cs]: huh, thought I already did that... I'll do it once I get home

Comment (AtomCrafty) [reply SampleGame/SampleGame.cs]: Oh I see, visual studio did some automatic formatting

**Comment by AtomCrafty** [comment]
This PR is blocking everything else I'm working on, so it'd be great if we could get it merged soon :)

**Comment by smoogipoo** [comment]
In regards to the review:

image

This is being triggered during the transition from fullscreen to windowed.

**Comment by AtomCrafty** [comment]
Not sure how I missed that...
Must have been introduced in the last commit, when I removed some seemingly unrelated checks from OnMove.

**Pull request merged**